src/main/java/io/strimzi/kafka/bridge/http/HttpSinkBridgeEndpoint.java

```diff
@@ -130,11 +130,15 @@ private void doSeek(JsonObject bodyAsJson) {
             if (done.succeeded()) {
                 HttpUtils.sendResponse(routingContext.response(), HttpResponseStatus.NO_CONTENT.code(), null, null);
             } else {
+                HttpResponseStatus statusCode = HttpResponseStatus.INTERNAL_SERVER_ERROR;
+                if (done.cause() instanceof IllegalStateException) {
```

Comment:
I'm surprised that the Apache `KafkaConsumer` doesn't have a dedicated exception for this. We can't really be 100% sure that all ISEs from `seek()` have as their cause the reasons we expect here. But matching on the exception message seems fragile and what other choice is there? So I'm OK with this, but have some qualms about it.

Reply:
I agree Tom ... but this is what we have from the native client right now :-(